cpp/include/cudf/io/orc.hpp

```diff
@@ -58,9 +58,9 @@ class orc_reader_options {
   // List of individual stripes to read (ignored if empty)
   std::vector<std::vector<size_type>> _stripes;
   // Rows to skip from the start;
-  size_type _skip_rows = 0;
-  // Rows to read; -1 is all
-  size_type _num_rows = -1;
+  uint64_t _skip_rows = 0;
```

Comment:
Should we use an `int64_t` instead of a `uint64_t` for this? If we don't absolutely need the extra power of 2 then it might be better to stick with a signed integer since we do that for `size_type`.

Reply:
The ORC files store the number of rows as uint64.
I understand that it's unlikely that an ORC file with over int64::max rows would ever exist. The idea is to match the specs (hoping to at some point add a header with ORC types that we can use in places like this).

Reply:
I was not aware that this was matching the spec. After some offline discussion, the general consensus on the team was that it was better to match the spec here, we should just add some comments indicating that is why we use this type.